src/lazy/encoder/binary/v1_1/value_writer.rs

```diff
@@ -402,3 +493,160 @@ impl<'value, 'top: 'value> ValueWriter for BinaryAnnotatedValueWriter_1_1<'value
+            (IonType::String, &[0xEB, 5]),
+            (IonType::Symbol, &[0xEB, 6]),
+            (IonType::Blob, &[0xEB, 7]),
+            (IonType::Clob, &[0xEB, 8]),
+            (IonType::List, &[0xEB, 9]),
+            (IonType::SExp, &[0xEB, 10]),
+            (IonType::Struct, &[0xEB, 11]),
+        ];
+        for (ion_type, expected_encoding) in test_cases {
+            encoding_test(
+                |writer: &mut LazyRawBinaryWriter_1_1<&mut Vec<u8>>| {
+                    writer.write(Null(*ion_type))?;
+                    Ok(())
+                },
+                expected_encoding,
+            )?;
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn write_bools() -> IonResult<()> {
+        let test_cases: &[(bool, &[u8])] = &[(true, &[0x5E]), (false, &[0x5F])];
+        for (value, expected_encoding) in test_cases {
+            encoding_test(
+                |writer: &mut LazyRawBinaryWriter_1_1<&mut Vec<u8>>| {
+                    writer.write(*value)?;
+                    Ok(())
+                },
+                expected_encoding,
+            )?;
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn write_ints() -> IonResult<()> {
+        let test_cases: &[(i64, &[u8])] = &[
+            (0, &[0x50]),
+            (-1, &[0x51, 0xFF]),
+            (1, &[0x51, 0x01]),
+            (100, &[0x51, 0x64]),
+            (-100, &[0x51, 0x9C]),
+            (127, &[0x51, 0x7F]),
+            (-127, &[0x51, 0x81]),
+            (128, &[0x52, 0x80, 0x00]),
+            (-128, &[0x51, 0x80]),
+            (
+                i64::MAX,
+                &[0x58, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x7F],
+            ),
+            (
+                i64::MIN,
+                &[0x58, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x80],
+            ),
+        ];
+        for (value, expected_encoding) in test_cases {
+            encoding_test(
+                |writer: &mut LazyRawBinaryWriter_1_1<&mut Vec<u8>>| {
+                    writer.write(*value)?;
+                    Ok(())
+                },
+                expected_encoding,
+            )?;
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn write_f32s() -> IonResult<()> {
+        let test_f64s: &[f32] = &[
+            1.0,
+            1.5,
+            -1.5,
+            10.0,
+            10.5,
+            -10.5,
+            f32::INFINITY,
+            f32::NEG_INFINITY,
+            f32::NAN,
+        ];
+        for value in test_f64s {
+            let mut expected_encoding = vec![0x5C];
+            expected_encoding.extend_from_slice(&value.to_le_bytes()[..]);
+            encoding_test(
+                |writer: &mut LazyRawBinaryWriter_1_1<&mut Vec<u8>>| {
+                    writer.write(value)?;
+                    Ok(())
+                },
+                expected_encoding.as_slice(),
+            )?;
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn write_f64s() -> IonResult<()> {
+        let test_f64s: &[f64] = &[
```

Comment:
Might be useful to have `&[(f64, &[u8])]` here so that you can have a case for `0.0` (and any other numbers that may eventually be supported using a smaller number of bytes).